### PR TITLE
[JENKINS-39520] ExtensionList.removeAll is unimplemented

### PR DESCRIPTION
**Pull request by jglick**
Complements https://github.com/jenkinsci/branch-api-plugin/pull/56.

@reviewbybees

### CODE REVIEW

File: core/src/main/java/hudson/ExtensionList.java [hunk 1]
```diff
@@ -204,6 +204,21 @@ public boolean remove(Object o) {
...
+    public boolean removeAll(Collection<?> c) {
+        try {
+            boolean removed = false;
+            for (Object o : c) {
+                removed |= removeSync(o);
+            }
+            return removed;
+        } finally {
+            if (extensions != null) {
+                fireOnChangeListeners();
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/ExtensionList.java]: I do not see a reason for invoking listeners if nothing has been removed. 🐜

Comment (jglick) [inline core/src/main/java/hudson/ExtensionList.java]: True, that could be optimized.

**Comment by oleg-nenashev** [comment]
@jglick Any plans to do it?
From what I see it's going to miss the weekly since there is no feedback CC @jenkinsci/code-reviewers

**Comment by jglick** [comment]
> it's going to miss the weekly

There is no rush; already working around this in `branch-api` anyway.

**Comment by oleg-nenashev** [comment]
🐝

**Comment by oleg-nenashev** [comment]
@reviewbybees done

**Review by KostyaSha — APPROVED** [review]

**Comment by KostyaSha** [comment]
does anything else missing from List APIs for this custom type?

**Comment by oleg-nenashev** [comment]
From what I see, it should be fine now. But still a point of failure since we may easily forget about new methods.

**Pull request merged**